experimental/manual/oss_fuzz_vuln_prompt.py

```diff
@@ -0,0 +1,203 @@
+    """
+  # If the range is a single commit, return the range as parent commit:commit.
+  if regression_range.find(":") == -1 and regression_range.find("..") == -1:
+    return f"{regression_range}~", regression_range
+
+  delimiter = ":" if ":" in regression_range else ".."
+  start_commit, end_commit = regression_range.split(delimiter)
+
+  return start_commit, end_commit
+
+
+def get_changeset_diff(repo_url, regression_range):
+  """Fetches the code diff for a given commit range in a Git repository."""
+  local_repo_path = get_local_repo_path(repo_url)
+
+  try:
+    if not os.path.exists(local_repo_path):
+      subprocess.run(["git", "clone", repo_url, local_repo_path],
+                     stdout=subprocess.DEVNULL,
+                     check=True)
+    else:
+      subprocess.run(["git", "pull"],
+                     cwd=local_repo_path,
+                     stdout=subprocess.DEVNULL,
+                     check=True)
+  except Exception as e:
+    raise RuntimeError(f"Error cloning/pulling repository {repo_url}: {e}")
+
+  try:
+    repo = Repo(local_repo_path)
+  except InvalidGitRepositoryError:
+    raise ValueError(f"Invalid Git repository path: {local_repo_path}")
+
+  start_commit, end_commit = get_commit_range(regression_range)
+  try:
+    diff = repo.git.diff(f"{start_commit}..{end_commit}")
+    return diff.encode('utf-8', 'replace').decode('utf-8')
+  except Exception as e:
+    raise RuntimeError(f"Error retrieving changeset diff: {e}")
+
+
+def get_file_content(repo_url, crash_revision, file_path):
+  """Fetches the content of a file in a Git repository."""
+  local_repo_path = get_local_repo_path(repo_url)
+  local_file_path = file_path[:].removeprefix('/src/')
+  local_file_path = local_file_path.split('/', 1)[-1]
+
+  try:
+    repo = Repo(local_repo_path)
+  except InvalidGitRepositoryError:
+    raise ValueError(f"Invalid git repository path: {local_repo_path}")
+
+  try:
+    commit = repo.commit(crash_revision)
+  except BadName:
+    print(f"Error: Commit hash '{crash_revision}' not found in repository.")
+    return None
+
+  try:
+    return commit.tree[local_file_path].data_stream.read().decode('utf-8')
+  except Exception:
+    print(f"Error: '{local_file_path}' not found at commit '{crash_revision}'.")
+    return None
+
+
+if __name__ == "__main__":
+  parser = argparse.ArgumentParser(
+      description="Generate a prompt for a security vulnerability.")
+  parser.add_argument("--repo_url", help="Path to the GitHub repository")
+  parser.add_argument(
+      "--regression_range",
+      help="Commit range in the format 'start_commit:end_commit'")
+  parser.add_argument("--crash_revision",
+                      help="Revision where the crash occurred")
+  parser.add_argument("--crash_stacktrace",
+                      help="File containing the crash stacktrace")
+
+  args = parser.parse_args()
+  os.makedirs(PROJECTS_DIR, exist_ok=True)
+
+  stacktrace = open(args.crash_stacktrace).read()
```

Comment:
nit: To ensure the opened file is closed
```python
with open(args.crash_stacktrace) as file:
    stacktrace = file.read()
```

Reply:
Done